src/main/java/de/php_perfect/intellij/ddev/error_reporting/SentryErrorReporter.java

```diff
@@ -0,0 +1,112 @@
+package de.php_perfect.intellij.ddev.error_reporting;
+
+import com.intellij.diagnostic.AbstractMessage;
+import com.intellij.diagnostic.IdeaReportingEvent;
+import com.intellij.ide.DataManager;
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.idea.IdeaLogger;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.diagnostic.ErrorReportSubmitter;
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent;
+import com.intellij.openapi.diagnostic.SubmittedReportInfo;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.util.Consumer;
+import io.sentry.Sentry;
+import io.sentry.SentryEvent;
+import io.sentry.protocol.SentryId;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+public class SentryErrorReporter extends ErrorReportSubmitter {
+    @NotNull
+    @Override
+    public String getReportActionText() {
+        return "Report to Author";
+    }
+
+    @Override
+    public boolean submit(IdeaLoggingEvent @NotNull [] events, @Nullable String additionalInfo, @NotNull Component parentComponent, @NotNull Consumer<? super SubmittedReportInfo> consumer) {
+        DataContext context = DataManager.getInstance().getDataContext(parentComponent);
+        Project project = CommonDataKeys.PROJECT.getData(context);
+
+        new Task.Backgroundable(project, "Sending error report") {
+            @Override
+            public void run(@NotNull ProgressIndicator indicator) {
+                List<SentryId> sentryIds = new ArrayList<>();
+                ThreadSync threadSync = new ThreadSync();
+
+                AtomicBoolean oneOrMoreFailed = new AtomicBoolean(false);
+
+                for (int i = 0; i < events.length; ++i) {
+                    SentryId sentryId = captureIdeaLoggingEvent(events[i], threadSync, oneOrMoreFailed);
+                    sentryIds.add(sentryId);
+
+                    indicator.setFraction((i + 1.0) / events.length);
+                }
+
+                announceReportSent(sentryIds, parentComponent);
+
+                SubmittedReportInfo.SubmissionStatus submissionStatus = oneOrMoreFailed.get() ?
+                        SubmittedReportInfo.SubmissionStatus.FAILED :
+                        SubmittedReportInfo.SubmissionStatus.NEW_ISSUE;
+                consumer.consume(new SubmittedReportInfo(submissionStatus));
+            }
+        }.queue();
```

Comment:
Instead of running .queue you can use com.intellij.openapi.progress.ProgressManager.run(). This gives more options like showing a progress bar (if configured to do so) and to cancel the process if it is stuck for whatever reason.